File: model/model-base/src/main/java/top/fifthlight/renderer/model/ByteBufferUtil.java
package top.fifthlight.renderer.model;

import org.jetbrains.annotations.NotNull;

import java.nio.ByteBuffer;

public class ByteBufferUtil {
    @NotNull
    public static ByteBuffer slice(ByteBuffer buffer, int index, int length) {
        return buffer.slice(index, length);
    }

    @NotNull
    public static ByteBuffer put(ByteBuffer buffer, int index, ByteBuffer src, int offset, int length) {
        return buffer.put(index, src, offset, length);
    }
}

File: model/model-base/src/main/kotlin/top/fifthlight/renderer/model/util/ByteBufferUtil.kt
package top.fifthlight.renderer.model.util

import top.fifthlight.renderer.model.ByteBufferUtil
import java.nio.ByteBuffer

// Workaround of Bazel rules-kotlin using JDK11 as build JDK
fun ByteBuffer.sliceWorkaround(index: Int, length: Int) = ByteBufferUtil.slice(this, index, length)
fun ByteBuffer.putWorkaround(index: Int, src: ByteBuffer, offset: Int, length: Int) =
    ByteBufferUtil.put(this, index, src, offset, length)
